tools/install.cmake

```diff
@@ -0,0 +1,157 @@
+file(GLOB MBEDTLS_FILES CONFIGURE_DEPENDS "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls/library/*.c")
+set_source_files_properties(
+    ${MBEDTLS_FILES}
+    PROPERTIES COMPILE_FLAGS
+    "-Wno-pedantic"
+)
+
+# Define any extra sources or includes outside the standard, making sure to use the same prefix.
+set(MQTT_EXTRA_SOURCES
+        ${MQTT_SERIALIZER_SOURCES})
+set(PKCS_EXTRA_SOURCES
+        "${MBEDTLS_FILES}"
+        "${COREPKCS11_LOCATION}/source/portable/posix/core_pkcs11_pal.c"
+        "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls_utils/mbedtls_utils.c"
+        "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls_utils/mbedtls_error.c")
+set(PKCS_EXTRA_INCLUDE_PRIVATE_DIRS
+    PRIVATE
+        "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls/include"
+        "${CORE_PKCS11_3RDPARTY_LOCATION}/mbedtls_utils")
+set(OTA_BACKENDS "OTA_HTTP" "OTA_MQTT")
+foreach(ota_backend ${OTA_BACKENDS})
+    set("${ota_backend}_EXTRA_INCLUDE_PUBLIC_DIRS"
+        ${OTA_INCLUDE_PUBLIC_DIRS})
+    set("${ota_backend}_EXTRA_INCLUDE_PRIVATE_DIRS"
+        ${OTA_INCLUDE_PRIVATE_DIRS})
+endforeach()
+
+set(ALL_CSDK_PUBLIC_HEADERS "")
+foreach(library_prefix ${LIBRARY_PREFIXES})
+    # Create the library target.
+    if(DEFINED "${library_prefix}_SOURCES")
+        string(TOLOWER "aws_iot_${library_prefix}" library_name)
+        add_library("${library_name}"
+        ${${library_prefix}_EXTRA_SOURCES}
+        ${${library_prefix}_SOURCES})
+    else()
+        continue()
+    endif()
+
+    # Add any extra includes defined for the library.
+    if(DEFINED "${library_prefix}_EXTRA_INCLUDE_PUBLIC_DIRS")
+        target_include_directories("${library_name}"
+                        PUBLIC ${${library_prefix}_EXTRA_INCLUDE_PUBLIC_DIRS})
+    endif()
+    if(DEFINED "${library_prefix}_EXTRA_INCLUDE_PRIVATE_DIRS")
+        target_include_directories("${library_name}"
+                        PRIVATE ${${library_prefix}_EXTRA_INCLUDE_PRIVATE_DIRS})
+    endif()
+
+    # Allow a path to a custom config header to be passed through a CMake flag.
+    set(config_prefix "${library_prefix}")
+    if(";${OTA_BACKENDS};" MATCHES ";${library_prefix};")
+        set(config_prefix "OTA")
+    endif()
+    if(DEFINED "${config_prefix}_CUSTOM_CONFIG_PATH")
+        target_include_directories("${library_name}"
+                                    PRIVATE ${${config_prefix}_CUSTOM_CONFIG_PATH})
+    else()
+        target_compile_definitions("${library_name}" PRIVATE -D${config_prefix}_DO_NOT_USE_CUSTOM_CONFIG)
+        # PKCS11 requires a config so include the one from the demos by default.
+        if(${config_prefix} STREQUAL "PKCS")
+            target_include_directories("${library_name}" PRIVATE
+                                        ${DEMOS_DIR}/pkcs11/common/include
+                                        ${LOGGING_INCLUDE_DIRS})
+            target_compile_definitions("${library_name}" PUBLIC
+                                       -DMBEDTLS_CONFIG_FILE="mbedtls_config.h")
```

Comment:
If PKCS11 requires a config file, then the user should be required to pass it instead of the one from our demos being used (imo).

I think it would be important to support selective library installation so that the users that do not want to install PKCS11 and also don't want to provide custom configs for any of the other libraries they want, they don't have to be forced to provide custom config for PKCS11

Reply:
I think it's fine if it is using the configs from the demo as those configs are specific to POSIX. I am supporting selective library installation though.